src/components/pip/pip.tsx

```diff
@@ -19,36 +21,66 @@ interface PIPComponentOwnProps {
   hide: () => void;
   onSideBySideSwitch?: () => void;
   onInversePIP?: () => void;
+  dragAndSnapManager?: DragAndSnapManager;
 }
 interface PIPComponentConnectProps {
-  guiClientRect?: {height: number; width: number};
+  guiClientRect?: GuiClientRect;
   playerHover?: boolean;
 }
+interface PIPComponentState {
+  isDragging: boolean;
+}
 type PIPComponentProps = PIPComponentOwnProps & PIPComponentConnectProps;
 @connect(mapStateToProps, utils.bindActions(reducers.shell.actions))
-export class Pip extends Component<PIPComponentProps> {
-  ref = createRef();
+export class Pip extends Component<PIPComponentProps, PIPComponentState> {
+  videoContainerRef = createRef<HTMLDivElement>();
+  pipContainerRef = createRef<HTMLDivElement>();
+  constructor(props: PIPComponentProps) {
+    super(props);
+    this.state = {
+      isDragging: false
+    };
+  }
 
   componentDidMount() {
-    this.ref.current.appendChild(this.props.childPlayer.getView());
+    const {childPlayer, dragAndSnapManager} = this.props;
+    this.videoContainerRef?.current?.appendChild(childPlayer.getView());
+    if (dragAndSnapManager && this.pipContainerRef.current) {
+      dragAndSnapManager.setDraggableContainer(this.pipContainerRef.current);
+      dragAndSnapManager.onStartDrag(this._onStartDrag);
+      dragAndSnapManager.onStopDrag(this._onStopDrag);
+    }
+  }
+
+  componentDidUpdate() {
+    const {dragAndSnapManager, guiClientRect} = this.props;
+    if (dragAndSnapManager && guiClientRect) {
+      dragAndSnapManager.setGuiClientRect(guiClientRect);
+    }
+  }
+
+  private _handleClick = (fn: Function) => (e: Event) => {
+    e.stopPropagation();
+    this.props.dragAndSnapManager?.destroy();
```

Comment:
each click does a destory?

Reply:
yes, we don't need dragAndSnapManager for PIPminimized and SbS